### PR TITLE
refactor: Support light mode in nav

### PR DESCRIPTION
**Pull request by EnMod**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-refactor-light-mode-nav-hashicorp.vercel.app) 🔎
- [Product Pages Preview link](https://dev-portal-git-refactor-light-mode-nav-hashicorp.vercel.app/terraform) 🔎
- [Asana task](https://app.asana.com/0/1121504896220179/1207832908083301/f) 🎟️

## 🗒️ What

This PR refactors the nav and relevant children components to support light mode.

## 🤷 Why

This comes after a design team recommendation while working on other changes to the nav.

## 🛠️ How

My approach here was to leverage our existing tokens' ability to change based on the site theme, and use those tokens instead of hardcoded values in key areas. In some instances (e.g.: the `.siteLogo` SVG on the homepage and the logo on product pages), I added `path` fill overrides to change the SVGs' color when the theme changes.

## 📸 Design Screenshots

There is no established spec for light mode at present; the preview has been approved by design.

## 🧪 Testing

- [x] Open the preview link.
- [x] Go to the bottom of the page and change the theme to Dark.
- [x] Ensure text in the nav is legible, and contrasts with its background.
- [x] Ensure the site logo in the top-left contrasts with its background.
- [x] Switch the theme to Light, and perform the same validation steps.
- [x] Navigate to the product pages preview link, and perform the same validation steps.

## 💭 Anything else?

This work does change which text colors show up when switching to dark mode, but now the colors should be more aligned with our general design system.

### CODE REVIEW

File: src/components/navigation-header/navigation-header.module.css [hunk 1]
```diff
@@ -105,24 +93,24 @@
...
 
-	@nest html[data-theme='dark'] & {
+	/* @nest html[data-theme='dark'] & {
 		color: var(--token-color-foreground-strong);
 
 		&:hover, &[aria-expanded='true'] {
 			border-color: var(--token-color-border-strong);
 		}
-	}
+	} */
```
Comment (nandereck) [inline src/components/navigation-header/navigation-header.module.css:115]: Why do we need to comment this code out?

**Review by nandereck — APPROVED** [review]
LGTM!

File: src/components/navigation-header/navigation-header.module.css (hunk 1 above)
Comment (EnMod) [reply src/components/navigation-header/navigation-header.module.css:115]: Oh, I was testing whether we needed that code, and left that in before review 😅 I'll double-check what's needed from it and remove in the next commit if I can (I should be able to) :+1:

Comment (EnMod) [reply src/components/navigation-header/navigation-header.module.css]: @nandereck Removed the comments as of this commit (they were safe to remove):
eaf25cf

**Pull request merged**